### PR TITLE
Give OPDS importers an API class (PP-501)

### PR DESCRIPTION
**Pull request by jonathangreen**
## Description

Previously OPDS importers were a special case, they had no API class, there there was a special code path for checkouts and fulfillment taken. Now they are treated like every other license source, with their own `BaseCirculationAPI`.

## Motivation and Context

This is another step along the path to removing external integrations from license source integrations.

## How Has This Been Tested?

- Running tests

## Checklist

- [X] I have updated the documentation accordingly.
- [X] All new and existing tests passed.

### CODE REVIEW

**Comment by tdilauro** [comment]
I started a review of this today and plan to finish it up tomorrow.

**Review by tdilauro — COMMENTED** [review]
This looks really good! Just a few really small issues.

File: tests/fixtures/odl.py [hunk 1]
```diff
@@ -33,29 +33,35 @@ class MonkeyPatchedODLFixture:
...
+    def _queue_response(self, status_code, headers={}, content=None):
         self.responses.insert(0, MockRequestsResponse(status_code, headers, content))
 
+    @staticmethod
     def _get(self, url, headers=None):
         self.requests.append([url, headers])
         response = self.responses.pop()
         return HTTP._process_response(url, response)
 
+    @staticmethod
```
[the remark this answers is not in the document]
Comment (jonathangreen) [reply tests/fixtures/odl.py:48]: I added a comment to clarify.

File: tests/fixtures/odl.py [hunk 2]
```diff
@@ -33,29 +33,35 @@ class MonkeyPatchedODLFixture:
...
         self.monkeypatch = monkeypatch
 
-
-@pytest.fixture(scope="function")
-def monkey_patch_odl(monkeypatch) -> MonkeyPatchedODLFixture:
-    """A fixture that patches the ODLAPI to make it possible to intercept HTTP requests for testing."""
-
-    def queue_response(self, status_code, headers={}, content=None):
+    @staticmethod
+    def _queue_response(self, status_code, headers={}, content=None):
```
[the remark this answers is not in the document]
Comment (jonathangreen) [reply tests/fixtures/odl.py]: I added a comment to clarify.

**Comment by jonathangreen** [comment]
Rebased on main, and made code review changes.

**Review by tdilauro — APPROVED** [review]
Looks good! 🚀

**Pull request merged**